src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs

```diff
@@ -230,5 +282,82 @@ public void ConnectHeaders_SupplyHeaders_ToDownloadArraySizeSameAsNumberOfHeader
+            ChainedHeaderTree cht = ctx.CreateChainedHeaderTree();
+            ChainedHeader chainTip = ctx.ExtendAChain(10);
+            cht.Initialize(chainTip, true);
+
+            List<BlockHeader> listOfExistingHeaders = ctx.ChainedHeaderToList(chainTip, 10);
+            cht.ConnectNewHeaders(2, listOfExistingHeaders);
+            ChainedHeader peerTwoTip = cht.GetChainedHeaderByPeerId(2);
+
+            // Peer B / Peer 2 is on the best chain.
+            peerTwoTip.HashBlock.Should().BeEquivalentTo(chainTip.HashBlock);
+            peerTwoTip.ChainWork.Should().BeEquivalentTo(chainTip.ChainWork);
+
+            // After that peer C / Peer 3 presents an alternative chain with a forkpoint below B's Tip.
+            BlockHeader forkedBlock = ctx.GetNewBlock(chainTip.Previous.Previous).Header;
+            cht.ConnectNewHeaders(3, new List<BlockHeader>() { forkedBlock });
+            ChainedHeader peerThreeTip = cht.GetChainedHeaderByPeerId(3);
+            ChainedHeader fork = chainTip.FindFork(peerThreeTip);
+            fork.Height.Should().BeLessThan(peerTwoTip.Height);
+
+            // After that peer A / PEER_ONE prolongs C's chain.
+            const int numberOfBlocksToExtend = 7;
+            ChainedHeader peerOneTip = ctx.ExtendAChain(numberOfBlocksToExtend, peerThreeTip);
+            List<BlockHeader> listOfPeerOnesHeaders = ctx.ChainedHeaderToList(peerOneTip, numberOfBlocksToExtend + 2 /* (forked 2 back) */);
+
+            // Sends all headers from the fork point
+            fork.HashBlock.Should().BeEquivalentTo(listOfPeerOnesHeaders[0].GetHash());
+
+            // With few valid and one invalid.
+            foreach (var header in listOfPeerOnesHeaders.GetRange(3, 6))
+            {
+                ctx.ChainStateMock.Setup(x => x.MarkBlockInvalid(header.GetHash(), null));
+            }
+
+            cht.ConnectNewHeaders(1, listOfPeerOnesHeaders);
+
+            // Peer A / PEER_ONE's fork should now have the longest chain
+            peerOneTip.Height.Should().BeGreaterThan(peerThreeTip.Height);
+        }
+
+        /// <summary>
+        /// Issue 9 @ Create chained header tree component #1321
+        /// Check that everything is always consumed.
+        /// </summary>
+        [Fact]
+        public void ConnectHeaders_MultiplePeers_CheckEverythingIsConsumed()
+        {
+            var ctx = new TestContext();
+            ChainedHeaderTree cht = ctx.CreateChainedHeaderTree();
+            ChainedHeader chainTip = ctx.ExtendAChain(11);
+            cht.Initialize(chainTip, true);
+            List<BlockHeader> listOfNewHeaders = ctx.ChainedHeaderToList(chainTip, 11);
+
+            List<BlockHeader> peer1Headers = listOfNewHeaders.GetRange(0, 6);
+            ConnectNewHeadersResult connectedNewHeadersPeer1 = cht.ConnectNewHeaders(1, peer1Headers);
+
+            // 5 unprocessed headers
+            Assert.True(chainTip.Height - connectedNewHeadersPeer1.Consumed.Height == 5);
```

Comment:
use a comparison to 5 rather than assert true - it will give a better output on failure - perhaps 
```
var unprocessedHeaderCount = chainTip.Height - connectedNewHeadersPeer1.Consumed.Height
unprocessedHeaderCount.ShouldBe(5)
//OR Assert.Equal(5, unprocessedHeaderCount);
```